src/main/webapp/app/components/tokenInputGroups/TokenExtendButton.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { ReactNode } from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row
+} from 'react-bootstrap';
+import { Token } from 'app/shared/api/generated/API';
+import classnames from 'classnames';
+import { observable, action } from 'mobx';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { notifyError, notifySuccess } from 'app/shared/utils/NotificationUtils';
+import client from 'app/shared/api/clientInstance';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+type TokenExtendButtonProps = {
+  token: Token;
+};
+
+@observer
+export default class TokenExtendButton extends React.Component<
+  TokenExtendButtonProps
+> {
+  @observable selectedDay: Date;
+
+  constructor(props: TokenExtendButtonProps) {
+    super(props);
+  }
+
+  @autobind
+  @action
+  extendExpirationDate(token: Token, newDate: string) {
```

Comment:
So basically this will just be a simple view component without data logic ties to it.